### PR TITLE
Updating to Scala 2.13.1 and bug fixes

### PR DESCRIPTION
**Pull request by kiroco12**
This PR updates part of the modules to Scala 2.13.1, but keeps mantaining the 2.12.10 version for compatibility with sbt-buildinfo. Also, fixes a problem with the number of completed exercises, for the people which made older exercises what now are deleted. Still have to fix a deprecation with the old collection API of Scala 2.12 in the compiler module.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -103,14 +105,15 @@ lazy val client = (project in file("client"))
...
+    scalacOptions += "-Ymacro-annotations",
     libraryDependencies ++= Seq(
-      "io.monix"      %%% "monix"     % v('monix),
-      "org.typelevel" %%% "cats-core" % v('catsversion),
-      "com.lihaoyi" %%% "scalatags" % v('scalatags) xscalajs,
-      "org.scala-js" %%% "scalajs-dom" % v('scalajsdom),
-      "be.doeraene" %%% "scalajs-jquery" % v('scalajsjquery) xscalajs,
-      "com.lihaoyi" %%% "upickle" % v('upickle),
-      "com.lihaoyi" %%% "utest"   % v('utest) % Test
+      "io.monix"      %%% "monix"     % v(Symbol("monix")),
```
Comment (juanpedromoreno) [inline build.sbt]: Let's replace the symbol usage by standard strings

File: build.sbt [hunk 2]
```diff
@@ -119,12 +122,13 @@ lazy val `evaluator-client` = (project in file("eval-client"))
...
-      "io.circe"      %% "circe-generic"       % v('circeversion),
-      "org.scalatest" %% "scalatest"           % v('scalatest) % Test
-    )
+      "org.http4s"    %% "http4s-blaze-client" % v("http4s"),
+      "org.http4s"    %% "http4s-circe"        % v("http4s"),
+      "io.circe"      %% "circe-core"          % v("circeversion"),
+      "io.circe"      %% "circe-generic"       % v("circeversion"),
+      "org.scalatest" %% "scalatest"           % v("scalatest") % Test
+    ),
+    crossScalaVersions := Seq(scala_212, scala_213)
```
Comment (juanpedromoreno) [inline build.sbt]: We don't need to cross-compile this project, having Scala 2.13 would be enough

File: build.sbt [hunk 3]
```diff
@@ -135,11 +139,12 @@ lazy val definitions = (project in file("definitions"))
...
-      "org.scalatest"              %% "scalatest"                 % v('scalatest),
-      "org.scalacheck"             %% "scalacheck"                % v('scalacheckversion),
-      "com.github.alexarchambault" %% "scalacheck-shapeless_1.14" % v('scalacheckshapeless)
-    )
+      "org.typelevel"              %% "cats-core"                 % v("catsversion"),
+      "org.scalatest"              %% "scalatest"                 % v("scalatest"),
+      "org.scalacheck"             %% "scalacheck"                % v("scalacheckversion"),
+      "com.github.alexarchambault" %% "scalacheck-shapeless_1.14" % v("scalacheckshapeless")
+    ),
+    crossScalaVersions := Seq(scala_212, scala_213)
```
Comment (juanpedromoreno) [inline build.sbt]: Same here

File: build.sbt [hunk 4]
```diff
@@ -177,11 +191,13 @@ lazy val compiler = (project in file("compiler"))
 lazy val `sbt-exercise` = (project in file("sbt-exercise"))
   .settings(name := "sbt-exercise")
   .settings(
-    scalaVersion := "2.12.10",
+    scalaVersion := scala_212,
```
Comment (juanpedromoreno) [inline build.sbt]: Shouldn't be Scala 2.13?

Comment (kiroco12) [reply build.sbt]: There's a dependency (`sbt-buildinfo`) which isn't updated to Scala 2.13 yet, so the plugin can't be updated to Scala 2.12.

File: build.sbt (hunk 2 above)
Comment (kiroco12) [reply build.sbt]: Libraries with dependencies used in the `sbt-exercise` project need a version in Scala 2.12. That's why cross-compilation.

**Pull request merged**